src/System.IO.Pipes/src/System/IO/Pipes/NamedPipeServerStream.Windows.cs

```diff
@@ -39,6 +39,20 @@ public sealed partial class NamedPipeServerStream : PipeStream
                 throw new ArgumentOutOfRangeException(nameof(pipeName), SR.ArgumentOutOfRange_AnonymousReserved);
             }
 
+            PipeSecurity pipeSecurity = null;
+
+            if ((options & PipeOptions.CurrentUserOnly) != 0)
+            {
+                SecurityIdentifier identifier = GetCurrentUser();
+                PipeAccessRule rule = new PipeAccessRule(identifier, PipeAccessRights.ReadWrite, AccessControlType.Allow);
```

Comment:
Windows should be creating a default DACL for the named pipe if we don't specify one, can you compare the default one with the one created when the rule is added? Just to be double sure...

Reply:
The default DACL will give full control to more users than the current one:
> The ACLs in the default security descriptor for a named pipe grant full control to the LocalSystem account, administrators, and the creator owner. They also grant read access to members of the Everyone group and the anonymous account.

Also in the docs it says that if we want to prevent any other user to connect to the pipe we should add the Sid to the DACL:
> To prevent remote users or users on a different terminal services session from accessing a named pipe, use the logon SID on the DACL for the pipe. The logon SID is used in run-as logons as well; it is the SID used to protect the per-session object namespace.

I just compared the rules with and without my change and that can be seen. Without my change it will have 4 rules involving the administrators, owner and others. With my change the Pipe will have only one rule set specifically to the owner's SID with ReadWrite access.